signaler/signaler.go

```diff
@@ -6,20 +7,17 @@ import (
 	"syscall"
 )
 
-var s = make(chan os.Signal, 1)
-
-func init() {
-	sigs := []os.Signal{
-		os.Interrupt,
-		os.Kill,
-		syscall.SIGTERM,
-		syscall.SIGABRT,
-	}
-	signal.Notify(s, sigs...)
+// WaitForInterrupt blocks until a termination signal is received and returns it.
+// It registers a temporary channel, unregisters it via signal.Stop() and returns the received signal.
+func WaitForInterrupt() os.Signal {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, getPlatformSignals()...)
+	s := <-c
+	signal.Stop(c) // unregister to avoid keeping channel referenced/registered
+	return s
 }
 
-// WaitForInterrupt waits until a os.Signal is
-// received and returns the result
-func WaitForInterrupt() os.Signal {
-	return <-s
+// getPlatformSignals returns the appropriate signals to listen for on the current platform.
+func getPlatformSignals() []os.Signal {
+	return []os.Signal{os.Interrupt, syscall.SIGTERM}
 }
```

Comment:
Am I missing something, or ... aren't we always returning the same values?
So ... This function can evaporate and the signal slice can move to L14.